### PR TITLE
DB-10184 Add SYSIBM.SYSKEYCOLUSE as a system view

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by jyuanca — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptToAddSysKeyColUseViewInSYSIBM.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+public class UpgradeScriptToAddSysKeyColUseViewInSYSIBM extends UpgradeScriptBase {
+    public UpgradeScriptToAddSysKeyColUseViewInSYSIBM(SpliceDataDictionary sdd, TransactionController tc) {
+        super(sdd, tc);
+    }
+
+    @Override
+    protected void upgradeSystemTables() throws StandardException {
+        sdd.createKeyColumnUseViewInSysIBM(tc);
+
+        SpliceLogUtils.info(LOG, "Catalog upgraded: updated syscolumns view in SYSIBM schema");
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptToAddSysKeyColUseViewInSYSIBM.java:30]: Please use the right view name in the log message.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSCONSTRAINTSRowFactory.java [hunk 2]
```diff
@@ -616,4 +618,68 @@ protected int getConstraintType(ExecRow row) throws StandardException{
...
+            "  '' AS PERIOD    -- indicates whether the column is the start or end column for the BUSINESS_TIME period, default blank\n" +
+            "FROM\n" +
+            "  (SELECT SC.CONSTRAINTNAME, SC.SCHEMAID, SC.TABLEID, K.CONGLOMERATEID\n" +
+            "     FROM SYS.SYSCONSTRAINTS SC, SYS.SYSKEYS K   -- get all unique constraint conglomerate IDs\n" +
+            "     WHERE K.CONSTRAINTID = SC.CONSTRAINTID\n" +
+            "   UNION\n" +
+            "   SELECT SC.CONSTRAINTNAME, SC.SCHEMAID, SC.TABLEID, PK.CONGLOMERATEID\n" +
+            "     FROM SYS.SYSCONSTRAINTS SC, SYS.SYSPRIMARYKEYS PK\n" +
+            "     WHERE PK.CONSTRAINTID = SC.CONSTRAINTID) C, -- get all primary key conglomerate IDs\n" +
+            "  SYSVW.SYSSCHEMASVIEW S,\n" +
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSCONSTRAINTSRowFactory.java:673]: I think we can use sys.syschemas here instead. I'm considering two reasons: 1) So far, schema restriction hasn't been applied to all the views in sysibm schema yet, so we can be consistent here. 2) the schemasview in your view definition is used only in the second UNION branch, so the schema restriction is not applied in a complete way.

**Review by yxia92 — CHANGES_REQUESTED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSCONSTRAINTSRowFactory.java (hunk 2 above)
Comment (ascend1) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSCONSTRAINTSRowFactory.java:673]: Good to know, I change it to `SYS.SYSSCHEMAS` in the latest commit.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptToAddSysKeyColUseViewInSYSIBM.java (hunk 1 above)
Comment (ascend1) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeScriptToAddSysKeyColUseViewInSYSIBM.java:30]: Done.

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by yxia92 — APPROVED** [review]
+1

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**